api/proto/sign/v1/sign_service.proto

```diff
@@ -11,13 +11,16 @@ service SignService {
   // Sign record using keyless OIDC based provider or using PEM-encoded private key with an optional passphrase
   rpc Sign(SignRequest) returns (SignResponse);
 
+  // Push a pre-generated signature to storage for a record
+  rpc PushSignature(PushSignatureRequest) returns (PushSignatureResponse);
```

Comment:
IMO this may not be the best place to handle storage ops. Maybe this should be moved to the store API